ArduCopter/crash_check.cpp

```diff
@@ -53,6 +58,65 @@ void Copter::crash_check()
     }
 }
 
+// crash_check - disarms motors if a crash has been detected
+// crashes are detected by the vehicle being more than 20 degrees beyond it's angle limits continuously for more than 1 second
+// called at MAIN_LOOP_RATE
+void Copter::thrust_loss_check()
+{
+    static uint16_t thrust_loss_counter;  // number of iterations vehicle may have been crashed
+
+    // exit immediately if motor failure handling already engaged
+    if (motors->get_thrust_boost()) {
+        return;
+    }
+
+    // return immediately if disarmed
+    if (!motors->armed() || ap.land_complete) {
+        thrust_loss_counter = 0;
+        return;
+    }
+
+    // check for desired angle over 15 degrees
+    // todo: add thrust angle to AC_AttitudeControl
+    const Vector3f angle_target =  attitude_control->get_att_target_euler_cd();
+    if ((fabs(angle_target.x) > THRUST_LOSS_CHECK_ANGLE_DEVIATION_CD) || (fabs(angle_target.y) > THRUST_LOSS_CHECK_ANGLE_DEVIATION_CD)) {
```

Comment:
I missed it, but CI fails because these are `fabs` instead of `fabsf`.

Reply:
txs!  fabs replaced with fabsf.